compiler/rustc_hir_analysis/src/hir_ty_lowering/dyn_compatibility.rs

```diff
@@ -221,6 +244,14 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                             if !projection_bounds.contains_key(&key) {
                                 projection_bounds.insert(key, (pred, supertrait_span));
                             }
+                            if tcx.is_lang_item(
+                                pred.skip_binder().def_id(),
+                                rustc_hir::LangItem::DerefTarget,
+                            ) {
+                                // We need to fill in a `Receiver<Target = ?>` bound because
```

Comment:
This is a bit self-evident, but it's missing the more important part that a supertrait like `Deref: Receiver<Target = Self::Target>` does not automatically elaborate into a projection bound for `Reciever`'s `Target` type.